payments-core/src/main/java/com/stripe/android/googlepaylauncher/GooglePayPaymentMethodLauncher.kt

```diff
@@ -0,0 +1,212 @@
+package com.stripe.android.googlepaylauncher
+
+import android.os.Parcelable
+import androidx.activity.ComponentActivity
+import androidx.activity.result.ActivityResultLauncher
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.lifecycleScope
+import com.stripe.android.googlepaylauncher.GooglePayPaymentMethodLauncher.Result
+import com.stripe.android.model.PaymentMethod
+import com.stripe.android.paymentsheet.DefaultGooglePayRepository
+import com.stripe.android.paymentsheet.GooglePayRepository
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.launch
+import kotlinx.parcelize.Parcelize
+
+/**
+ * A drop-in class that presents a Google Pay sheet to collect a customer's payment.
+ *
+ * When successful, will return a [PaymentMethod] via [Result.Completed.paymentMethod].
+ */
+internal class GooglePayPaymentMethodLauncher internal constructor(
+    lifecycleScope: CoroutineScope,
+    private val config: Config,
+    private val googlePayRepositoryFactory: (GooglePayEnvironment) -> GooglePayRepository,
+    private val readyCallback: ReadyCallback,
+    private val activityResultLauncher: ActivityResultLauncher<GooglePayPaymentMethodLauncherContract.Args>
+) {
+    private var isReady = false
+
+    /**
+     * Constructor to be used when launching [GooglePayPaymentMethodLauncher] from an Activity.
+     *
+     * @param activity the Activity that is launching the [GooglePayPaymentMethodLauncher]
+     *
+     * @param readyCallback called after determining whether Google Pay is available and ready on
+     * the device. [present] may only be called if Google Pay is ready.
+     *
+     * @param callback called with the result of the [GooglePayPaymentMethodLauncher] operation
+     */
+    internal constructor(
+        activity: ComponentActivity,
+        config: Config,
+        readyCallback: ReadyCallback,
+        callback: ResultCallback
+    ) : this(
+        activity.lifecycleScope,
+        config,
+        googlePayRepositoryFactory = {
+            DefaultGooglePayRepository(
+                activity.application,
+                it
+            )
+        },
+        readyCallback,
+        activity.registerForActivityResult(
```

Comment:
we should probably also mention in the doc that `GooglePayPaymentMethodLauncher` can only be created in `Activity#onCreate`